libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -74,3 +74,34 @@ void ConstantEvaluator::endVisit(Literal const& _literal)
 	if (!_literal.annotation().type)
 		m_errorReporter.fatalTypeError(_literal.location(), "Invalid literal value.");
 }
+
+void ConstantEvaluator::endVisit(Identifier const& _identifier)
+{
+	if (depth > DEPTH_LIMIT)
+	{
+		m_errorReporter.fatalTypeError(_identifier.location(), "Cyclic constant definition.");
+	}
+
+	VariableDeclaration const* variableDeclaration = dynamic_cast<VariableDeclaration const*>(_identifier.annotation().referencedDeclaration);
+	if (!variableDeclaration)
+		return;
+	if (!variableDeclaration->isConstant())
+	{
+		m_errorReporter.fatalTypeError(_identifier.location(), "Identifier must be declared constant.");
```

Comment:
We consider `fatalTypeError` to be throwing so it doesn't need to have an else branch.